### PR TITLE
Add Update Endpoints and Fix other endpoints for new schema

### PR DESCRIPTION
**Pull request by JonMike8**
#175, #140, #36 (i think)

### CODE REVIEW

File: app/api/v1/antenna/validate.ts [hunk 1]
```diff
@@ -1,32 +1,38 @@
 import { z } from 'zod';
 
 export type Antenna = {
-  id: string;
+  id: number;
```
Comment (OwenTruong) [inline app/api/v1/antenna/validate.ts:4]: I think the type is the following:

```typescript
export type Antenna = {
  id: string;
  name: string;
  hostname: string;
  model: string;
  modelName: string;
  frequency: number;
  playground_frequency: number;
  latitude: string;
  longitude: string;
  azimuth: number;
  typeAntenna: 0 | 1 | 2;
  antenna_status: string;
  cpu: number | null;
  ram: number | null;
};```

File: app/api/v1/antenna/validate.ts [hunk 2]
```diff
@@ -1,32 +1,38 @@
...
+  typeantenna: number;
+  antenna_status: string;
+  cpu: number;
+  ram: number;
 };
 
 export function isAntenna(antenna: unknown): antenna is Antenna {
   const antennaSchema = z.object({
-    id: z.string(),
+    id: z.number(),
```
Comment (OwenTruong) [inline app/api/v1/antenna/validate.ts:22]: Also for the schema:

```typescript
const antennaSchema = z.object({
  id: z.string(),
  name: z.string(),
  hostname: z.string(),
  model: z.string(),
  modelName: z.string(),
  frequency: z.number().int().gte(0),
  playground_frequency: z.number().int().gte(0),
  latitude: z.string(),
  longitude: z.string(),
  azimuth: z.number().int().gte(0).lt(360),
  typeAntenna: z.union([z.literal(0), z.literal(1), z.literal(2)]),
  antenna_status: z.string(),
  cpu: z.nullable(z.number().gte(0)),
  ram: z.nullable(z.number().gte(0)),
});```

Comment (OwenTruong) [reply app/api/v1/antenna/validate.ts:22]: id is a string because mongodb stores id as a hex number

**Review by OwenTruong — CHANGES_REQUESTED** [review]

File: app/api/v1/antenna/validate.ts (hunk 1 above)
Comment (OwenTruong) [reply app/api/v1/antenna/validate.ts:4]: Actually not sure about cpu and ram. In the last meeting, we might have said to set it to 0

Comment (miguel-merlin) [reply app/api/v1/antenna/validate.ts:4]: @JonMike8 Can you change the schema to the one Owen provided please.

**Review by OwenTruong — APPROVED** [review]

File: app/api/v1/antenna/[id]/route.ts [hunk 3]
```diff
@@ -4,9 +4,9 @@ import { isAntenna } from '@/app/api/v1/antenna/validate';
 import { pool } from '@/app/api/v1/connection';
 import StatusError from '@/app/api/(utils)/StatusError';
 
-export async function DELETE(_: Request, context: { params: { id: string } }) {
+export async function DELETE(_: Request, context: { params: { id: number } }) {
   try {
-    if (context.params.id.trim().length == 0)
+    if (!context.params.id)
```
Comment (miguel-merlin) [inline app/api/v1/antenna/[id]/route.ts:9]: Nice!

File: app/api/v1/antenna/[id]/route.ts [hunk 4]
```diff
@@ -42,3 +42,57 @@ export async function DELETE(_: Request, context: { params: { id: string } }) {
...
   }
 }
+export async function PUT(
+  request: Request,
+  context: { params: { id: number } }
+) {
+  try {
+    if (!context.params.id) {
+      throw {
+        status: 500,
```
Comment (miguel-merlin) [inline app/api/v1/antenna/[id]/route.ts:52]: This should throw 400 status error (Bad request). 500 status code errors are reserved for internal server errors (Connection with database fail, etc)

File: app/api/v1/antenna/[id]/route.ts [hunk 5]
```diff
@@ -42,3 +42,57 @@ export async function DELETE(_: Request, context: { params: { id: string } }) {
...
+        message: 'Internal Error: Id is missing from parameter',
```
Comment (miguel-merlin) [inline app/api/v1/antenna/[id]/route.ts:53]: Follow up to previous comment: Change the error message to bad request, i.e. missing id

File: app/api/v1/antenna/[id]/route.ts [hunk 6]
```diff
@@ -42,3 +42,57 @@ export async function DELETE(_: Request, context: { params: { id: string } }) {
...
+      RETURNING *;
+      `;
+    const result = await client.query(query, [frequency, id]);
+
+    if (result.rowCount === 0) {
+      throw new StatusError(404, `Antenna with ${id} does not exist.`);
+    }
+    if (!isAntenna(result.rows[0])) {
+      throw new StatusError(
+        500,
```
Comment (miguel-merlin) [inline app/api/v1/antenna/[id]/route.ts:76]: Status code 400

File: app/api/v1/antenna/route.ts [hunk 7]
```diff
@@ -14,3 +17,53 @@ export async function GET() {
     return NextResponse.json({ message }, { status: 500 });
   }
 }
+
+// request body should be:
```
Comment (miguel-merlin) [inline app/api/v1/antenna/route.ts:21]: Remove unnecessary comments

File: app/api/v1/antenna/route.ts [hunk 8]
```diff
@@ -14,3 +17,53 @@ export async function GET() {
...
+      WHERE id = $2
+      RETURNING *
+      `;
+      const result = await client.query(query, [frequency, id]);
+      if (result.rowCount === 0) {
+        throw new StatusError(404, `Antenna with ${id} does not exist.`);
+      }
+      if (!isAntenna(result.rows[0])) {
+        throw new StatusError(
+          500,
```
Comment (miguel-merlin) [inline app/api/v1/antenna/route.ts:47]: Status code 400

**Review by miguel-merlin — APPROVED** [review]
Great Job!

**Comment by miguel-merlin** [comment]
@JonMike8 Do you want to do the merge? Or do you want me to do it?

**Comment by JonMike8** [comment]
@miguel-merlin i can do it, doesn't look like many conflicts

**Review by miguel-merlin — APPROVED** [review]

**Pull request merged**